app/controllers/search_controller.rb

```diff
@@ -0,0 +1,46 @@
+class SearchController < ApplicationController
+  before_action :authenticate_user
+
+  breadcrumb 'Search', :search_path, only: [:search]
+
+  def search
+    @q = search_term
+
+    offenders = SearchService.search_for_offenders(@q, active_caseload)
+    total = offenders.count
+
+    @offenders = get_slice_for_page(offenders, page)
+    @page_meta = create_page_meta(total, @offenders.count)
+  end
+
+private
+
+  # At this point @offenders contains ALL of the offenders at the prison that
+  # match the search term. We will only show 10 at a time, so there is no need
+  # to get the allcocated POM name for all of them.
+  def get_slice_for_page(offenders, page_number)
+    start = [(page_number - 1) * 10, 0].max
+    slice = offenders.slice(start, 10)
+
+    OffenderService.set_allocated_pom_name(slice, active_caseload)
```

Comment:
Sorry I don't get why this line here - can you explain?

Reply:
We need to set the POM name for the offenders, but rather than do it for all 600, we just do it for the 10 we are about to show on the current page.

Reply:
Will add a comment

Reply:
Yeah, nastily set_allocated_pom_name will edit in place, and then return the things it's just modified.  I think it's working because the list (ref) that is passed is having it's contents mutated.  Fixed in latest commit.